pkg/controller/controller.go

```diff
@@ -4,296 +4,157 @@
-
-		// temporary ZMS call to get the signed domain
-		signedDomain, err := zms.GetSignedDomain(domainName)
-		if err != nil {
-			log.Println("Error getting the signed domain from ZMS:", err)
-			continue
-		}
-
-		domainRBAC := m.ConvertAthenzPoliciesIntoRbacModel(signedDomain.Domain)
-		rbacCRs := c.rbacProvider.ConvertAthenzModelIntoIstioRbac(domainRBAC)
-
-		for _, v := range rbacCRs {
-			log.Printf("CustomResource: %s/%s/%s: ", v.Type, v.Namespace, v.Name)
-			log.Println("Contents: ", v.Spec)
-		}
 
-		// legacy
-		for _, role := range domain.Roles {
-			// ex: service.role.domain.service
-			namespace := util.DomainToNamespace(domainName)
-			roleName := strings.TrimPrefix(string(role.Role.Name), domainName+":role.service.role.")
-			serviceRole, exists := serviceRoleMap[roleName+"-"+namespace]
-			if !exists {
-				log.Println("Service role", roleName, "does not exist, creating...")
-				err := c.srMgr.CreateServiceRole(namespace, c.dnsSuffix, roleName, role.Policy)
-				if err != nil {
-					log.Println("Error creating service role:", err)
-					continue
-				}
-				log.Println("Created service role", roleName, "in namespace", namespace)
-				continue
-			}
-
-			log.Println("Service role", roleName, "already exists, updating...")
-			serviceRole.Processed = true
-			updated, err := c.srMgr.UpdateServiceRole(serviceRole.ServiceRole, c.dnsSuffix, roleName, role.Policy)
-			if err != nil {
-				log.Println("Error updating service role:", err)
-				continue
-			}
-			if updated {
-				log.Println("Updated service role", roleName, "in namespace", namespace)
-			} else {
-				log.Println("No difference found for service role", roleName, "in namespace", namespace,
-					"not updating")
-			}
-
-			if len(role.Role.Members) == 0 {
-				log.Println("Role", roleName, "has no members, skipping service role binding creation")
-				continue
-			}
-
-			serviceRoleBinding, exists := serviceRoleBindingMap[roleName+"-"+namespace]
-			if !exists {
-				err = c.srbMgr.CreateServiceRoleBinding(namespace, roleName, role.Role.Members)
-				if err != nil {
-					log.Println("Error creating service role binding:", err)
-					continue
-				}
-				log.Println("Created service role binding", roleName, "in namespace", namespace)
-				continue
-			}
-			log.Println("Service role binding", roleName, "already exists, updating...")
-			serviceRoleBinding.Processed = true
-			updated, err = c.srbMgr.UpdateServiceRoleBinding(serviceRoleBinding.ServiceRoleBinding,
-				namespace, roleName, role.Role.Members)
-			if err != nil {
-				log.Println("Error updating service role binding:", err)
-				continue
-			}
-
-			if updated {
-				log.Println("Updated service role binding", roleName, "in namespace", namespace)
-			} else {
-				log.Println("No difference found for service role binding", roleName, "in namespace", namespace,
-					"not updating")
-			}
-		}
+	if !exists {
+		return fmt.Errorf("athenz domain %s does not exist in cache", key)
```

Comment:
is this an error condition? we might have to delete the CRs for the Athenz domain if they were created at some point.

Reply:
we could add a TODO, to later add delete enqueue calls to the processing controller

Reply:
Added a todo comment here so we can handle it when integrating with the processing controller.